frontend/ui/elements/font_settings.lua

```diff
@@ -0,0 +1,68 @@
+local Device = require("device")
+local util = require("util")
+local _ = require("gettext")
+
+--[[ Font settings for desktop linux and mac ]]--
+
+local function getUserDir()
+    return os.getenv("HOME").."/.local/share/fonts"
+end
+
+local function canOpenUserFontDir()
+    return Device:canOpenLink() and util.pathExists(getUserDir())
+end
+
+local function usesSystemFonts()
+    return G_reader_settings:isTrue("desktop_uses_system_fonts")
+end
+
+-- System fonts are common in linux
+local function getSystemDir()
+    local path = "/usr/share/fonts"
+    if util.pathExists(path) then
+        return path
+    else
+        -- mac doesn't use ttf fonts
+        return nil
+    end
+end
+
+local FontSettings = {}
+
+function FontSettings:getPath()
+    if G_reader_settings:isTrue("desktop_uses_system_fonts") then
+        local sys_path = getSystemDir()
+        if sys_path ~= nil then
+            return getUserDir()..";"..sys_path
+        end
+    end
+    return getUserDir()
+end
+
+function FontSettings:getMenuTable()
+    return {
+        text = _("Font settings"),
+        sub_item_table = {
+            {
+                text = _("Enable system fonts"),
+                checked_func = usesSystemFonts,
+                callback = function()
+                    G_reader_settings:saveSetting("desktop_uses_system_fonts", not usesSystemFonts())
+                    local UIManager = require("ui/uimanager")
+                    local InfoMessage = require("ui/widget/infomessage")
+                    UIManager:show(InfoMessage:new{
+                        text = _("This will take effect on next restart.")
+                    })
+                end,
+            },
+            {
+                text = _("Open fonts dir"),
```

Comment:
```suggestion
                text = _("Open fonts folder"),
```

Or directory, but I think we (and most filemanagers) use folder.